cwf/k8s/cwf_operator/pkg/controller/hacluster/hacluster_controller.go

```diff
@@ -147,47 +152,73 @@ func (r *ReconcileHACluster) Reconcile(request reconcile.Request) (reconcile.Res
+			"max consecutive errors for failover",
+			hacluster.Spec.MaxConsecutiveActiveErrors,
+		)
+		hacluster.Status.ConsecutiveActiveErrors = hacluster.Status.ConsecutiveActiveErrors + 1
+		if hacluster.Status.ConsecutiveActiveErrors < hacluster.Spec.MaxConsecutiveActiveErrors {
+			updateErr := r.client.Status().Update(context.TODO(), hacluster)
+			if updateErr != nil {
+				reqLogger.Error(updateErr, "An error occurred while updating consecutive error total for active")
+			}
+			return reconcile.Result{RequeueAfter: retryPeriod}, updateErr
+		}
 	} else {
+		hacluster.Status.ConsecutiveActiveErrors = 0
 		reqLogger.Info("Fetched active health status", "health", activeHealth.Health.String(), "message", activeHealth.HealthMessage)
 	}
 	standbyHealth, err := r.getGatewayHealthStatus(standbyGateway, request.Namespace)
 	if err != nil {
+		hacluster.Status.ConsecutiveStandbyErrors = hacluster.Status.ConsecutiveStandbyErrors + 1
 		reqLogger.Error(err, "An error occurring while fetching standby health status")
 	} else {
+		hacluster.Status.ConsecutiveStandbyErrors = 0
 		reqLogger.Info("Fetched standby health status", "health", standbyHealth.Health.String(), "message", standbyHealth.HealthMessage)
 	}
 
 	failover := false
 	var initErr error
 	var updatedStatus magmav1alpha1.HAClusterStatus
 	if activeHealth.Health == protos.HealthStatus_HEALTHY {
-		updatedStatus, initErr = r.initCluster(activeGateway, standbyGateway, request.Namespace, hacluster.Status)
+		updatedStatus, initErr = r.initCluster(activeGateway, standbyGateway, request.Namespace, hacluster.Status, false)
 	} else if activeHealth.Health == protos.HealthStatus_UNHEALTHY && standbyHealth.Health == protos.HealthStatus_HEALTHY {
-		failover = true
 		reqLogger.Info("Promoting standby due to unhealthy active", "promoted active", standbyGateway)
 		hacluster.Status.ActiveInitState = magmav1alpha1.Uninitialized
 		hacluster.Status.StandbyInitState = magmav1alpha1.Uninitialized
-		updatedStatus, initErr = r.initCluster(standbyGateway, activeGateway, request.Namespace, hacluster.Status)
+		hacluster.Status.ConsecutiveActiveErrors = 0
+		hacluster.Status.ConsecutiveStandbyErrors = 0
+		updatedStatus, initErr = r.initCluster(standbyGateway, activeGateway, request.Namespace, hacluster.Status, true)
 	} else {
 		reqLogger.Info("Both gateways are detected to be unhealthy")
-		updatedStatus, initErr = r.initCluster(activeGateway, standbyGateway, request.Namespace, hacluster.Status)
+		// If both gateways are unhealthy, there is a chance that a restart
+		// occurred, rendering the gateway(s) uninitialized. As a precaution,
+		// update the status to uninitialized and re-init. If the gateways are
+		// already initialized, this will be a no-op
+		hacluster.Status.ActiveInitState = magmav1alpha1.Uninitialized
+		hacluster.Status.StandbyInitState = magmav1alpha1.Uninitialized
+		updatedStatus, initErr = r.initCluster(activeGateway, standbyGateway, request.Namespace, hacluster.Status, false)
```

Comment:
For readability, alias the conditionals so this can flow like

```
if isPrimaryHealthy {
  ...
} else if isStandbyHealthy {
  ...
} else {
  // both unhealthy
}
```